### PR TITLE
Reduce the amount federated seed workers

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:

The current default settings for the ShootState controller and the ControllerInstallationRequired Controller in the federated seed controller lead to ca. 280 workers being created.
By reducing the default settings and fixing a bug during worker creation, the amount of workers created is reduced to the amount of extensions watched per Seed - around 20 per Seed.

Also do not update the ShootState if the Extension State is null. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Adjust default concurrent-sync settings for the Gardenlet controllers `ShootState` and `ControllerInstallationRequired`  causing too many goroutines to be created. Also fixed a bug during worker creation to further reduce the amount of workers being created. 
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm
/cc @plkokanov @swilen-iwanow @vlvasilev 
/needs second-opinion

File: pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go [hunk 1]
```diff
@@ -83,6 +83,11 @@ func (s *shootStateControl) createShootStateSyncReconcileFunc(ctx context.Contex
 		purpose := extensionObject.GetExtensionSpec().GetExtensionPurpose()
 		state := extensionObject.GetExtensionStatus().GetState()
 
+		if state == nil {
+			s.log.Debugf("Skip Shoot's %s %s extension state with name %s - no state found.", shoot.Name, kind, name)
+			return reconcile.Result{}, nil
+		}
```
Comment (swilen-iwanow) [inline pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go]: I am a bit worried about this. As far as I remember I've also added such logic, but then we decided to remove the state from the ShootState if it set to nil in the Extension's state. @plkokanov, wdyt?

Comment (plkokanov) [reply pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go]: I guess we could change the logic a bit to work like this:
if the extension state in the shoot state was not nil, but in the current reconciliation it is nil, we should update the ShootState and delete the extension state from it [check this line](https://github.com/gardener/gardener/pull/2331/files/f304afd50eb3d162a025fac78f5f33bc1fd64b1c#diff-5fbd8d379e4dd574afdac838e68337f2R110). Otherwise we do not update if the state in the current reconciliation is nil.

Basically if we could add a check before calling CreateOrUpdate would be great imo.

Comment (rfranzke) [reply pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go]: Hm, this sounds unrelated to this PR. So probably @danielfoehrKn should revert this change for now and either he or you can open a new one?

Comment (danielfoehrKn) [reply pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go]: I would appreciate if @swilen-iwanow or @plkokanov could take over this issue / give it another thought as you are the experts on the topic. 
I ll revert the change.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by swilen-iwanow — APPROVED** [review]
lgtm
